docs/latest/design/tcp-udp-design.md

```diff
@@ -0,0 +1,36 @@
+# TCP&UDP Proxy Design 
+
+Even though most of the use cases for Envoy Gateway are at Layer-7, Envoy Gateway can also work at Layer-4 to proxy TCP 
+and UDP traffic. This document will explore the options we have when operating Envoy Gateway at Layer-4 and explain the 
+design decision.
+
+Envoy can work as a non-transparent proxy or a [transparent proxy](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/other_features/ip_transparency#arch-overview-ip-transparency-original-src-listener), 
+so ideally, Envoy Gateway should also be able to work in these two modes:
+
+## Non-transparent Proxy Mode
```

Comment:
if you are clubbing TCP and UDP as one, then you can't use `terminate` connection since its doesn't apply to UDP :)
so recommend either creating two separate sections for UDP and TCP